### PR TITLE
Implement repo.Describe()

### PR DESCRIPTION
**Pull request by nulltoken**
Now that https://github.com/libgit2/libgit2/pull/2592 is merged, let's make this available to LibGit2Sharp.

### CODE REVIEW

File: LibGit2Sharp/IRepository.cs [hunk 1]
```diff
@@ -1,6 +1,5 @@
 using System;
 using System.Collections.Generic;
-using LibGit2Sharp.Handlers;
```
Comment (nulltoken) [inline LibGit2Sharp/IRepository.cs]: This removal should be in a separate commit

File: LibGit2Sharp/IRepository.cs [hunk 2]
```diff
@@ -366,5 +365,7 @@ public interface IRepository : IDisposable
         /// <param name="options">If set, the options that control the status investigation.</param>
         /// <returns>A <see cref="RepositoryStatus"/> holding the state of all the files.</returns>
         RepositoryStatus RetrieveStatus(StatusOptions options);
+
+        string Describe(Commit commit, DescribeOptions options);
```
Comment (nulltoken) [inline LibGit2Sharp/IRepository.cs]: missing xml doc

File: LibGit2Sharp/Repository.cs [hunk 3]
```diff
@@ -1863,6 +1863,14 @@ private IEnumerable<string> RemoveStagedItems(IEnumerable<string> paths, bool re
             return removed;
         }
 
+        public string Describe(Commit commit, DescribeOptions options)
```
Comment (nulltoken) [inline LibGit2Sharp/Repository.cs]: missing xml doc

File: LibGit2Sharp/RepositoryExtensions.cs [hunk 4]
```diff
@@ -738,5 +738,10 @@ public static RepositoryStatus RetrieveStatus(this IRepository repository)
             Proxy.git_index_read(repository.Index.Handle);
             return new RepositoryStatus((Repository)repository, null);
         }
+
+        public static string Describe(this IRepository repository, Commit commit)
```
Comment (nulltoken) [inline LibGit2Sharp/RepositoryExtensions.cs]: missing xml doc

**Comment by nulltoken** [comment]
Although this lacks some xml doc, I think it's more or less ready for review

File: LibGit2Sharp/Core/Proxy.cs [hunk 5]
```diff
@@ -608,6 +608,68 @@ public static ConfigurationSafeHandle git_config_snapshot(ConfigurationSafeHandl
...
         #endregion
 
+        #region git_describe_
+
+        public static string git_describe_commit(
+            RepositorySafeHandle repo,
+            ObjectId committishId,
+            DescribeOptions options)
+        {
+            Ensure.ArgumentConformsTo(options.MinimumCommitIdAbbreviatedSize,
```
Comment (Therzok) [inline LibGit2Sharp/Core/Proxy.cs]: Ensure.ArgumentInt32 ?

File: LibGit2Sharp/IRepository.cs (hunk 1 above)
Comment (nulltoken) [inline LibGit2Sharp/IRepository.cs]: Fixed

**Comment by nulltoken** [comment]
>> +        public static string git_describe_commit(
>> +            RepositorySafeHandle repo,
>> +            ObjectId committishId,
>> +            DescribeOptions options)
>> +        {
>> +            Ensure.ArgumentConformsTo(options.MinimumCommitIdAbbreviatedSize,

>Ensure.ArgumentInt32 ?

@Therzok Better?

**Comment by Therzok** [comment]
Yeah, better. I had removed the comment since it only had ResultInt32 in Ensure. Probably going to be useful later on, that's why I asked.

File: LibGit2Sharp/DescribeOptions.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        /// </para>
+        /// </summary>
+        public DescribeOptions()
+        {
+            Strategy = DescribeStrategy.Default;
+            MinimumCommitIdAbbreviatedSize = 7;
+        }
+
+        /// <summary>
+        /// The kind of references that will be elligible as reference points.
```
Comment (jamill) [inline LibGit2Sharp/DescribeOptions.cs]: elligible => eligible

Comment (nulltoken) [inline LibGit2Sharp/DescribeOptions.cs]: Fixed

**Comment by nulltoken** [comment]
Xml doc has been finalized.

Ready for final review.

**Comment by nulltoken** [comment]
Rebased. Good to go?
